### PR TITLE
Conditional ES5 Browser Polyfill Loading

### PR DESCRIPTION
**Pull request by clydin**
- no need to manually import and manage individual ES2015 polyfills required by Angular
  - ES2015 polyfills are only loaded by browsers that require them
  - Controllable via a new `es5BrowserSupport` option; no behavior change if option is not enabled
  - `es5BrowserSupport` is enabled in newly generated projects
  - allows for custom polyfill solutions by not using the new option
  - works in conjunction with conditional JIT polyfills and will load ES2015 specific JIT polyfills as needed
  - Saves ~56KB on native ES2015 browsers

### CODE REVIEW

File: tests/legacy-cli/e2e/tests/basic/scripts-array.ts [hunk 1]
```diff
@@ -54,6 +54,7 @@ export default function () {
     // index.html lists the right bundles
     .then(() => expectFileToMatch('dist/test-project/index.html', oneLineTrim`
       <script type="text/javascript" src="runtime.js"></script>
+      <script type="text/javascript" src="es2015-polyfills.js" nomodule></script>
```
Comment (mathiasbynens) [inline tests/legacy-cli/e2e/tests/basic/scripts-array.ts]: Drive-by comment: can we drop the `type="text/javascript"` please? It’s redundant: https://mathiasbynens.be/notes/html5-levels#type-attributes

Comment (clydin) [reply tests/legacy-cli/e2e/tests/basic/scripts-array.ts]: Yes.  A refactor and cleanup of the index generation is planned that will also address this.

File: packages/schematics/angular/application/files/src/polyfills.ts [hunk 2]
```diff
@@ -18,31 +18,9 @@
...
-// import 'core-js/es6/math';
-// import 'core-js/es6/string';
-// import 'core-js/es6/date';
-// import 'core-js/es6/array';
-// import 'core-js/es6/regexp';
-// import 'core-js/es6/map';
-// import 'core-js/es6/weak-map';
-// import 'core-js/es6/set';
-
 /** IE10 and IE11 requires the following for NgClass support on SVG elements */
```
Comment (alan-agius4) [inline packages/schematics/angular/application/files/src/polyfills.ts]: Any reason why this was not moved as well?

Comment (clydin) [reply packages/schematics/angular/application/files/src/polyfills.ts]: It requires an additional package install. It is also optional and DOM related rather than ES2015 related.

**Comment by tomgruszowski** [comment]
Are there any docs on how to enable this?

**Comment by belf2601** [comment]
How can we enable on current projects ?

**Comment by diicar** [comment]
It's enabled by defauly, just need to Update cli to v7.3.x I believe.

**Comment by neilhem** [comment]
Add `es5BrowserSupport: true` to angular.json options

**Comment by maxailloud** [comment]
How can I fix the ```Schema validation failed with the following errors:
  Data path "" should NOT have additional properties(es5BrowserSupport).``` error?
I just added the option to my build option.

**Comment by clydin** [comment]
@maxailloud The project most likely has an older version of `@angular-devkit/build-angular` (for this feature 0.13.0+ is needed).

Also, the team is more than happy to help when possible, but please avoid using a PR for these type of discussions as they are not generally monitored.

**Comment by maxailloud** [comment]
Oh my bad, thought it was an issue.
And btw don't know why but i still had the 0.12.4 version of it, even after executing ng update @angular/cli.

Thank you and sorry again.

**Comment by paulogr** [comment]
And on your previously created projects, don't forget to remove those es6 polyfills imports from polyfills.ts file for this feature be real effective!

File: packages/angular_devkit/build_angular/src/angular-cli-files/models/es2015-polyfills.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+import 'core-js/es6/parse-float';
+import 'core-js/es6/number';
+import 'core-js/es6/math';
+import 'core-js/es6/string';
+import 'core-js/es6/date';
+import 'core-js/es6/array';
+import 'core-js/es6/regexp';
+import 'core-js/es6/map';
+import 'core-js/es6/weak-map';
+import 'core-js/es6/set';
```
Comment (DmitryEfimenko) [inline packages/angular_devkit/build_angular/src/angular-cli-files/models/es2015-polyfills.js]: I might be mistaken, but it's my understanding that polyfills are moved into an internal file. I often find that I also need to import some of the es7 modules (for example to support `Object.entries`) into polifyls. Having them in an internal file won't allow me adding additional polifylls only required for legacy browsers. It'd be my preference if there was a new file generated by CLI and included in the app: `polifyls.legacy.ts` or something like that.
